sqlmesh/core/engine_adapter/_typing.py

```diff
@@ -8,6 +8,8 @@
 if t.TYPE_CHECKING:
     import pyspark
     import pyspark.sql.connect.dataframe
+    from bigframes.session import Session as BigframeSession  # noqa
+    from bigframes.dataframe import DataFrame as BigframeDataFrame
```

Comment:
Are these imports safe? What if you don't have `bigframes` installed?

Reply:
It works the same way as SQLMesh does without Snowpark in Python 3.12

The if t.TYPE_CHECKING means the code is never hit at runtime, its just for the type checker. And the type checker is okay if an import doesn't exist. At worst it complains. Hence why the whole test suit passes even though bigframes is not installed. 

cc @georgesittas

Reply:
Gotcha, makes sense.